src/overlays/actors/ovl_En_Niw/z_en_niw.c

```diff
@@ -21,61 +48,1192 @@ const ActorInit En_Niw_InitVars = {
+}
+
+void func_80AB63A8(EnNiw* this, GlobalContext* globalCtx) {
+    if (this->actor.bgCheckFlags & 1) {
+        if (this->actor.velocity.y < 0.0f) {
+            this->unk_2AC = this->unk_2B8 = this->actor.posRot.pos.x;
+            this->unk_2B0 = this->unk_2BC = this->actor.posRot.pos.y;
+            this->unk_2B4 = this->unk_2C0 = this->actor.posRot.pos.z;
+            this->timer5 = this->timer4 = this->unk_29E = 0;
+
+            this->unk_26C[7] = this->unk_26C[5] = this->unk_26C[6] = this->unk_26C[8] = this->actor.speedXZ =
+                this->unk_2FC = this->unk_300 = 0.0f;
+
+            this->actionFunc = func_80AB6570;
+            return;
+        }
+    }
+
+    func_80AB5BF8(this, globalCtx, 2);
+}
+
+void func_80AB6450(EnNiw* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+
+    if (this->actor.xzDistFromLink < 30.0f) {
+        if (fabsf(this->actor.posRot.pos.y - player->actor.posRot.pos.y) < 5.0f) {
+            this->timer6 = 100;
+            this->actor.gravity = -2.0f;
+            this->actionFunc = func_80AB7290;
+            return;
+        }
+    }
+
+    if (Actor_HasParent(&this->actor, globalCtx)) {
+        this->actor.gravity = -2.0f;
+        Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+        this->sfxTimer1 = 30;
+        this->path = 0;
+        this->timer4 = 30;
+        this->actor.flags &= ~1;
+        this->actor.speedXZ = 0.0f;
+        this->actionFunc = func_80AB6BF8;
+        return;
+    }
+
+    func_8002F434(&this->actor, globalCtx, 0, 25.0f, 10.0f);
+    func_80AB5BF8(this, globalCtx, 1);
+}
+
+void func_80AB6570(EnNiw* this, GlobalContext* globalCtx) {
+    f32 pi;
+    s32 pad;
+    f32 fVar2 = Math_Rand_CenteredFloat(100.0f);
+    f32 fVar3 = Math_Rand_CenteredFloat(100.0f);
+    s16 iVar1;
+
+    if (this->actor.params != 0x000A) {
+        if (Actor_HasParent(&this->actor, globalCtx)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+            this->sfxTimer1 = 30;
+            this->path = 0;
+            this->timer4 = 30;
+            this->actor.flags &= ~1;
+            this->actor.speedXZ = 0.0f;
+            this->actionFunc = func_80AB6BF8;
+            return;
+        }
+
+        func_8002F580(&this->actor, globalCtx);
+    } else {
+        if (this->path != 0) {
+            this->unk_2A6 = 1;
+
+            if (this->sfxTimer3 == 0) {
+                Audio_PlayActorSound2(&this->actor, NA_SE_EV_CHICKEN_CRY_M);
+                this->sfxTimer3 = 100;
+            }
+
+            this->unk_2A0 = Math_Rand_ZeroFloat(1.99f);
+            this->actor.speedXZ = 4.0f;
+            this->unk_300 = 0.0f;
+            this->unk_2FC = 0.0f;
+            this->actionFunc = func_80AB6A38;
+            return;
+        }
+    }
+
+    iVar1 = 0;
+
+    if (this->timer5 != 0) {
+        if (Math_Rand_ZeroFloat(3.99f) < 1.0f) {
+            this->unk_2E6++;
+            this->unk_2E6 &= 1;
```

Comment:
```suggestion
            this->unk_2E6 %= 2;
```
may or may not match again

Reply:
Doesn't match here either